maven/WFLY-14934_Server_Plugin.adoc

```diff
@@ -0,0 +1,186 @@
+
+In addition to these 2 new goals, we are replacing the existing automatic download and unzip of wildfly-dist by a Galleon provisioning of 
+WildFly. This change impacts ```run```, ```start``` and ```execute-commands``` goals).
+ 
+==== Galleon related Plugin configuration items
+
+These configuration items are common to the ```provision``` and ```package``` goals. They are used to configure 
+Galleon to provision a server.
+
+* feature-packs: A list of feature-pack to install.
+* log-time: Whether to log provisioning time at the end.
+* off-line: Whether to use offline mode when the plugin resolves an artifact.
+* plugin-options: Arbitrary Galleon options used when provisioning the server.
+* provision-directory-name: By default the server is generated in ```target/server``` directory. 
+The plugin offers a configuration item to provide the name of a directory relative to the project ```target``` directory in which the server will be installed.
+* provisioning-file: The path to the provisioning.xml file to use, by default ```<project base dir>/galleon/provisioning``` is used. If some feature-packs have been set, 
+the provisioning file is being ignored.
+* record-state: Whether to record provisioned state in .galleon directory. When false, the file ```.wildfly-maven-plugin-provisioning.xml``` is generated in the server home directory.
+It contains Galleon provisioning informations used to provision this server.
+* server-configuration: A configuration contains the included and excluded Galleon layers to build the configuration.
+A configuration is identified by a ```model``` (default to ```standalone```) and a ```name``` (default to ```standalone.xml```).
+ 
+==== ```package``` goal specific configuration items
+
+These configuration items names, when applicable, are re-using the names that are in use in existing WildFly Maven plugin goals.
+
+* cli-sessions: A list of CLI execution session. Each session can contain: a list of CLI scripts, a list of commands, a list of properties files, 
+a list of java options, the ability to resolve or not expressions locally (false by default) and system properties for the CLI forked process. 
+NB: CLI commands are automatically executed in the context of an embedded server.
+* extra-server-content-dirs: A list of directories to copy content to the provisioned server inside a forked process.
+* filename: The name of the artifact to deploy inside the project build directory. By default a lookup is done to retrieve the primary artifact to deploy.
+* name: The name of the deployment. By default the file name is used.
+* runtime-name: The runtime name of the deployment. By default the file name is used. NB: In order to deploy 
+the deployment inside the root context, ```ROOT.war``` runtime name is to be used.
+* server-config: The server config to use when executing CLI scripts and deploying artifact. Is ```standalone.xml``` by default if no ```server-groups``` have 
```

Comment:
I'm not sure to understand what it would mean to run `mvn package` with a `server-groups`. 

The first use case for this goal is in the cloud context where WildFly always runs in standalone mode. I wonder if focusing on that use case might not be better (and add support for domain mode when we widen the scope of this plugin)

Reply:
This is an existing parameter that would be impacted by the `package` goal. It's not clear to me how it interact with the `server-configuration/model & name`

Reply:
@jmesnil , server-configuration/model and name is removed, I replaced by layers config element. We are keeping server-config with a simple default value "standalone.xml". domai nsupport being removed.